### PR TITLE
Someip input plugin

### PR DESCRIPTION
**Pull request by anthonypayne-GM**
Add an input plugin that can capture data from a SOME/IP network. The plugin has two primary use cases:

1. Subscribe to SOME/IP events. When an event notification is received, it will generate a data record and publish it to the data pipeline.
2. Send a SOME/IP request. When the SOME/IP response is received, the plugin will generate a data record.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [X] Example configuration file for the change
- [X] Debug log output from testing the change
- [X] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found[fb-in_someip-valgrind.txt](https://github.com/user-attachments/files/17547542/fb-in_someip-valgrind.txt)

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [N/A] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [N/A] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by anthonypayne-GM** [comment]
Attaching logs from running the in_some plugin against a test SOME/IP service.
[fb-in_someip-test-service.log](https://github.com/user-attachments/files/17547872/fb-in_someip-test-service.log)
[fluent-bit-in_someip.log](https://github.com/user-attachments/files/17547874/fluent-bit-in_someip.log)

File: cmake/plugins_options.cmake [hunk 1]
```diff
@@ -62,6 +62,7 @@ DEFINE_OPTION(FLB_IN_WINLOG                   "Enable Windows Log input plugin"
 DEFINE_OPTION(FLB_IN_WINDOWS_EXPORTER_METRICS "Enable windows exporter metrics input plugin" ON)
 DEFINE_OPTION(FLB_IN_WINEVTLOG                "Enable Windows EvtLog input plugin"           OFF)
 DEFINE_OPTION(FLB_IN_WINSTAT                  "Enable Windows Stat input plugin"             OFF)
+DEFINE_OPTION(FLB_IN_SOMEIP                   "Enable SOME/IP input plugin"                  OFF)
```
Comment (anthonypayne-GM) [inline cmake/plugins_options.cmake]: Looks like the rest of these plugins are in alphabetical order. Let's move this to line 49.5

File: lib/libsomeip-c/src/someip_wrapper.cc [hunk 2]
```diff
@@ -0,0 +1,856 @@
...
+                    handler(request_id, payload_bytes, payload_size);
+                }};
+            std::lock_guard<std::mutex> lock(context_mutex_);
+            application_->register_message_handler(method.service.service_id,
+                                                   method.service.instance_id,
+                                                   method.method_id,
+                                                   std::move(message_handler));
+        }
+
+        void SendResponse(const uint32_t request_id,
```
Comment (anthonypayne-GM) [inline lib/libsomeip-c/src/someip_wrapper.cc:444]: Add some documentation to this class.

File: plugins/in_someip/in_someip.c [hunk 3]
```diff
@@ -0,0 +1,847 @@
...
+ *
+ * @param data Pointer to the plugin context
+ * @param config not used
+ */
+static void in_someip_pause(void *data, struct flb_config *config)
+{
+    struct flb_someip *ctx = data;
+    (void) config;
+
+    // TODO: Should we pause subscription during this time?
```
Comment (anthonypayne-GM) [inline plugins/in_someip/in_someip.c]: REmove TODO

**Review by anthonypayne-GM — COMMENTED** [review]
Few things to clean up.

File: cmake/plugins_options.cmake (hunk 1 above)
Comment (anthonypayne-GM) [reply cmake/plugins_options.cmake]: Done in latest commit.

File: plugins/in_someip/in_someip_config.c [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+    if (an_rpc == NULL) {
+        flb_errno();
+        return -1;
+    }
+    an_rpc->service_id = service;
+    an_rpc->instance_id = instance;
+    an_rpc->method_id = method;
+    encoded_len = flb_sds_len(message);
+    if (encoded_len > 0) {
+        decoded_buffer = flb_sds_create_size(encoded_len);
```
Comment (GMLucasC) [inline plugins/in_someip/in_someip_config.c:113]: We aren't freeing this variable in the event of an error.

**Review by GMLucasC — COMMENTED** [review]
Looks very good. One potential memory leak found.

File: plugins/in_someip/in_someip_config.c (hunk 4 above)
Comment (anthonypayne-GM) [reply plugins/in_someip/in_someip_config.c:113]: Good catch. Will clean up at around line 120 and line 127

**Review by GMLucasC — APPROVED** [review]

**Pull request merged**